### ISSUE TITLE
[FEATURE REQUEST] Proxy rules for other proxy apps (e.g. shadowrocket) - is it possible

### ISSUE THREAD

**Issue by dlfnn** [SideStore/SideStore#368]
### Describe the feature

The proxy sidestore currently uses to refresh apps will force deactivate other proxies on the phone, which is bad for people in china mainland or other regions who need a proxy to bypass government firewall. By having a config file for e.g. shadowrocket, it might be easier to have both.

### Use cases

To be able to refresh sideloaded apps while also bypass government firewalls e.g. GFW. It would be especially helpfull for people in China who want to use a github source (which is partially banned by GFW)

**Comment by naturecodevoid** [comment]
Known issue, we are planning to PR wireguard or something to allow for us to change the VPN only when we need it.

However, it most likely won't be possible to use multiple VPNs at once (unless we rewrite the proxy using a native VPN type; those might allow multiple at once)

**Comment by dlfnn** [comment]
Can wireguard broadcast its proxy to a localhost port? 
If so might be able to make certain domains go through its port and others through shadowrocket's from shadowrocked config?

**Comment by naturecodevoid** [comment]
Yes, that might be possible

**Comment by Paris0-0** [comment]
Need this too, using other VPN apps that support wireguard protocol like shadowrocket or [Surge ](https://manual.nssurge.com/policy/wireguard.html) or  [Stash](https://stash.wiki/en/proxy-protocols/proxy-types#wireguard) and using the sample config file can't refresh SideStore app either. Only WireGuard iOS App can cooperate with SideStore to refresh.

**Comment by Spidy123222** [comment]
> Need this too, using other VPN apps that support wireguard protocol like shadowrocket or [Surge ](https://manual.nssurge.com/policy/wireguard.html) or  [Stash](https://stash.wiki/en/proxy-protocols/proxy-types#wireguard) and using the sample config file can't refresh SideStore app either. Only WireGuard iOS App can cooperate with SideStore to refresh.

There is some other apps that work like passpartout but do note that passpartout has a lot more battery drain for some reason.

**Comment by danaenayati** [comment]
Hello guys, We have this issue in Iran too. Anyone could find a solution for this?

**Comment by haomuch** [comment]
Yes we need this feature. And it's best to proxy sidestore refresh data traffic through shadowrocket on any Protocol other than wireguard, for example a single proxy rule is used.

**Comment by Mattease** [comment]
> 是的，我们需要这个功能。最好使用除 WireGuard 以外的任何协议，通过 Shadowrocket 代理 Sidestore 刷新数据流量，例如，可以使用单个代理规则。

现在已经可以了 测试版的新功能 不通过wireguard

**Comment by haomuch** [comment]
> > 是的，我们需要这个功能。最好使用除 WireGuard 以外的任何协议，通过 Shadowrocket 代理 Sidestore 刷新数据流量，例如，可以使用单个代理规则。
> 
> 现在已经可以了 测试版的新功能 不通过wireguard

我没有发现最新的sidestore测试版里如何配置这个功能，但是我知道Shadowrocket最新的测试版里加入了针对sidestore的本地反射代理功能，因此以后只需开启Shadowrocket就能刷新sidestore了，而且不会影响Shadowrocket的正常代理规则
